lib/logger/lib/logger.ex

```diff
@@ -695,49 +680,63 @@ defmodule Logger do
           :ok | {:error, :noproc} | {:error, term}
   def bare_log(level, chardata_or_fun, metadata \\ []) do
     case __should_log__(level) do
-      :error -> :ok
-      info -> __do_log__(info, chardata_or_fun, metadata)
+      nil -> :ok
+      level -> __do_log__(level, chardata_or_fun, Map.new(metadata))
     end
   end
 
   @doc false
+  def __should_log__(:warn) do
+    # TODO: Uncomment this
+    # bare_log(:warning, "`:warn` level is deprecated, you should use `:warning`")
+
+    __should_log__(:warning)
+  end
+
   def __should_log__(level) when level in @levels do
-    case __metadata__() do
-      {true, pdict} ->
-        case Logger.Config.log_data(level) do
-          {:discard, _config} -> :error
-          {mode, config} -> {level, mode, config, pdict}
-        end
+    %{level: primary_level} = :logger.get_primary_config()
 
-      {false, _} ->
-        :error
+    if enabled?(self()) and compare_levels(level, primary_level) != :lt do
+      level
     end
   end
 
   @doc false
-  def __do_log__({level, mode, config, pdict}, chardata_or_fun, metadata)
-      when is_list(metadata) do
-    %{utc_log: utc_log?, truncate: truncate} = config
-    metadata = [pid: self()] ++ into_metadata(metadata, pdict)
-
-    case normalize_message(chardata_or_fun, metadata) do
-      {message, metadata} ->
-        tuple = {Logger, truncate(message, truncate), Logger.Utils.timestamp(utc_log?), metadata}
-
-        try do
-          notify(mode, {level, Process.group_leader(), tuple})
-          :ok
-        rescue
-          ArgumentError -> {:error, :noproc}
-        catch
-          :exit, reason -> {:error, reason}
-        end
+  def __do_log__(:warn, chardata_or_fun, metadata) do
```

Comment:
This clause should not be necessary because we already translate `:warn` to `:warning` on `__should_log__`?